--- a/configure.ac
+++ b/configure.ac
@@ -578,7 +578,11 @@
 # pthreads is harmless otherwise.
 CFG_LIBS="$LIBS $CFG_LIBS"
 _MY_LDLIBS_CHECK_OPT(CFG_LIBS, -lpthread)
-_MY_LDLIBS_CHECK_OPT(CFG_LIBS, -latomic)
+if test "$CFG_ENABLE_PARTIAL_STATIC" = "yes"; then
+  _MY_LDLIBS_CHECK_OPT(CFG_LIBS, -Wl,-Bstatic -latomic -Wl,-Bdynamic)
+else
+  _MY_LDLIBS_CHECK_OPT(CFG_LIBS, -latomic)
+fi
 
 # Check libraries for MingW
 _MY_LDLIBS_CHECK_OPT(CFG_LIBS, -lbcrypt)
